compiler/rustc_const_eval/messages.ftl

```diff
@@ -128,15 +128,19 @@ const_eval_frame_note_inner = inside {$where_ ->
 
 const_eval_frame_note_last = the failure occurred here
 
+const_eval_incompatible_arg_types =
+    calling a function whose {$arg_idx ->
+        [1] 1st
+        [2] 2nd
+        *[other] {$arg_idx}th
+    } parameter has type {$callee_ty} passing argument of type {$caller_ty}
```

Comment:
Could you rephrase this so we don't use an ordinal (1st/2nd/3rd/4th)? I think it's fine to just say "argument #n" or something.

AFAICT right now it's not right for 3 (3rd) or any other numbers that end in 1/2/3.